Makefile

```
@@ -1,7 +1,7 @@
 NAME := $(shell grep 'name =' Cargo.toml | head -n 1 | cut -d'"' -f2)
 VERSION := $(shell grep '^version =' Cargo.toml | cut -d'"' -f2)
 ARCH ?= $(shell uname -m)
-TARGET_ARCH ?= $(ARCH)-unknown-linux-gnu
+TARGET_ARCH ?= $(shell rustc -vV | sed -n 's/host: //p')
```

This change may produce undesirable results if someone forgets to set both `ARCH` and `TARGET_ARCH` when trying to cross-compile using the Makefile. Ideally we should set the `ARCH` based on the user-provided `TARGET_ARCH` to prevent this issue:

```suggestion
TARGET_ARCH ?= $(shell rustc -vV | sed -n 's/host: //p')
ARCH := $(shell echo "$(TARGET_ARCH)" | cut -d'-' -f1)
```